browser/ui/tabs/split_view_tab_tile_data.h

```diff
@@ -78,45 +88,47 @@ class SplitViewBrowserData {
    private:
     void RunCurrentClosureIfNeededAndReplaceWith(OnTabDragEndedClosure&& other);
 
-    raw_ptr<SplitViewBrowserData> data_;
+    raw_ptr<SplitViewTabTileData> data_;
 
     base::ScopedClosureRunner closure_;
   };
   [[nodiscard]] OnTabDragEndedClosure TabDragStarted();
 
   void TabsWillBeAttachedToNewBrowser(const std::vector<tabs::TabHandle>& tabs);
-  void TabsAttachedToNewBrowser(SplitViewBrowserData* target_data);
+  void TabsAttachedToNewBrowser(SplitViewTabTileData* target_data);
 
  private:
-  friend class SplitViewBrowserDataBrowserTest;
+  friend class SplitViewTabTileDataBrowserTest;
   friend class SplitViewTabStripModelAdapterBrowserTest;
 
-  FRIEND_TEST_ALL_PREFIXES(SplitViewBrowserDataBrowserTest,
+  FRIEND_TEST_ALL_PREFIXES(SplitViewTabTileDataBrowserTest,
                            BreakTile_WithNonExistingTabIsError);
-  FRIEND_TEST_ALL_PREFIXES(SplitViewBrowserDataBrowserTest,
+  FRIEND_TEST_ALL_PREFIXES(SplitViewTabTileDataBrowserTest,
                            TileTabs_WithAlreadyTiledTabIsError);
-  FRIEND_TEST_ALL_PREFIXES(SplitViewBrowserDataBrowserTest, FindTile);
+  FRIEND_TEST_ALL_PREFIXES(SplitViewTabTileDataBrowserTest, FindTabTile);
 
-  std::vector<TabTile>::iterator FindTile(const tabs::TabHandle& tab);
-  std::vector<TabTile>::const_iterator FindTile(
+  std::vector<TabTile>::iterator FindTabTile(const tabs::TabHandle& tab);
+  std::vector<TabTile>::const_iterator FindTabTile(
       const tabs::TabHandle& tab) const;
 
-  void Transfer(SplitViewBrowserData* other, std::vector<TabTile> tiles);
+  // When tabs are attached to another browser window and they are tiled tabs,
+  // create tab tiles to that browser by using |tab_tiles|.
+  void Transfer(SplitViewTabTileData* other, std::vector<TabTile> tab_tiles);
 
   std::unique_ptr<SplitViewTabStripModelAdapter> tab_strip_model_adapter_;
 
-  std::vector<TabTile> tiles_;
-  std::vector<TabTile> tiles_to_be_attached_to_new_window_;
+  std::vector<TabTile> tab_tiles_;
+  std::vector<TabTile> tab_tiles_to_be_attached_to_new_window_;
 
   // As UI is likely to read more frequently than insert or delete, we cache
```

Comment:
I'm confused about what exactly this is a cache of, what would be the alternative way to look this up and what is the index?

Reply:
This is for finding `TabTile` of `TabHandle` from `tab_tiles` quickly.
index here is the `tab_tiles` vector index.
With this `TabHandle` and index map, we can get `TabTile` that includes `TabHandle` from `tab_tiles_` quickly. W/o this, we have to iterate all tiles in `tab_tiles_` whenever we want to check current tab is in tile or not. or to get tab's tab tile. Updated this comments.